File: src/CouchChefBackend/CouchChefBLL/DTOs/IngredientDTO.cs
namespace CouchChefBLL.DTOs;

public class IngredientDTO
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int? ImageId { get; set; }
    public ImageDTO? ImageDTO { get; set; }
    public string? Description { get; set; }
    public float Protein { get; set; }
    public float Fat { get; set; }
    public float Carbs { get; set; }
    public float? Calories { get; set; }

    public bool IsValid()
    {
        return this.Carbs + this.Protein + this.Fat <= 100;
    }
}

File: src/CouchChefFrontend/src/app/entities/ingredient.dto.ts
import { GetImageDTO, PostImageDTO } from '@entities/image.dto';

export interface GetIngredientDTO {
    id: number,
    name: string,
    imageId: number,
    getImageDTO: GetImageDTO,
    description: string,
    protein: number,
    fat: number,
    carbs: number,
    calories: number
}

export interface PostIngredientDTO {
    id: number,
    name: string,
    postImageDTO: PostImageDTO,
    description: string,
    protein: number,
    fat: number,
    carbs: number
}